### DISCUSSION TITLE
Date Parsing Filter Operator

### DISCUSSION THREAD

**Discussion by andrewgoz** [TiddlyWiki/TiddlyWiki5#7676] [Ideas]
As part of my work developing a plugin for the ApexCharts library, I found a need to parse date strings in filters. I wanted to turn "12-Aug-2023" into a TiddlyWiki DateValue "20230812000000000". I was a bit surprised to be unable to find something like that already implemented, so I wrote one. Examples in the comments:
```
/*\
title: Parse Date Filter Operator
type: application/javascript
module-type: filteroperator

Filter operator for parsing a string as a Date or Unix Timestamp
and returning a DateValue (YYYY0MM0DD0hh0mm0ss0XXX)

If the current timezone is UTC+8:
"[[12-Aug-2023]parsedate[]]" -> "20230812000000000"
"[[12-Aug-2023]parsedate[UTC]]" -> "20230811160000000"
"[[20230812000000000]format:date[TIMESTAMP]parsedate[TIMESTAMP]]" -> "20230812000000000"
"[[20230812000000000]format:date[TIMESTAMP]parsedate[UTCTIMESTAMP]]" -> "20230812080000000"

\*/
/*jslint node: true, browser: true */
/*global $tw: false */
"use strict";

/*
Export our filter function
*/
exports.parsedate = function(source,operator,options) {
	var results = [];
	source(function(tiddler,title) {
		let utc = operator.operand.indexOf('UTC') != -1;
		let ts = operator.operand.indexOf('TIMESTAMP') != -1;
		let val = ts ? parseInt(title) : Date.parse(title);
		let d = new Date();
		d.setTime(val - ((utc == ts) ? d.getTimezoneOffset() * 60000 : 0));
		results.push($tw.utils.stringifyDate(d));
	});
	return results;
};
```
Unless I've missed something, I think this would be a good addition to the core code.

**Comment by ericshulman** [comment]
This functionality is already provided by the core code.
TW5.2.4 added `format:date[TIMESTAMP]` and
TW5.3.0 added `format:timestamp[...TW date format string...]`.

Both of these are demonstrated in the 4th example in https://tiddlywiki.com/#format%20Operator%20(Examples).

You might also take a look at: https://tiddlytools.com/#TiddlyTools%2FTime%2FParseDate is part of my extensive set of TiddlyTools "Time" add-ons (see https://tiddlytools.com/#TiddlyTools%2FTime%2FInfo for full documentation).

Note that my TiddlyTools `parsedate[]` filter not only converts TWCore date formats, but also any valid IS08601 or RFC2822 formatted datetime string.  For example: `{{{ [[July 24th, 1962]parsedate[YYYY0MM0DD]] }}}` returns `19620724`

enjoy,
-e

**Reply by andrewgoz** [reply]
Thanks. I had completely overlooked `format:timestamp`. That left the text parsing which your filter operator does nicely.

I'll ditch my code and stop re-inventing the wheel, at least for this.
